### ISSUE TITLE
Linux embeddings of Flutter does not support multi-touch!

### ISSUE THREAD

**Issue by caijw** [flutter/flutter#52202]
## Use case

Running flutter apps on an ubuntu machine with touchscreen.
I found that multi-touch fails.
After reading the [source code](https://github.com/flutter/engine/blob/master/shell/platform/glfw/flutter_glfw.cc#L298) of flutter engine for platform embedding, i found that it is because that GLFW does not support multi-touch.
GLFW now just supports [mouse input](https://www.glfw.org/docs/latest/input_guide.html#input_mouse), and as a result , flutter apps running on ubuntu mathine do not support multi-touch.

## Proposal

Sorry I don't hava suitable proposal for this problem (maybe transforming the pointer evnet directly to flutter enging form libinput is a good way), I am here for asking for help.
It's there any solution for this problem ? Or what i can do for it ?
Supporting multi-touch on ubuntu is really important for me!

**Comment by dnfield** [comment]
Sounds like this is blocked by https://github.com/glfw/glfw/issues/42

**Comment by caijw** [comment]
It seems that it will be long after for  GLFW to support multi-touch.
Is there any suggestion on supporting multi-touch on linux platform?
@dnfield

**Comment by stuartmorgan** [comment]
I don't think there's any point trying to solve this using GLFW since work on https://github.com/flutter/flutter/issues/30729 has started, and that will replace the GLFW implementation. We should instead consider it blocked on that issue.

**Comment by maxjoehnk** [comment]
What has to be done for this to work now that the gtk based shell is implemented as far as I can tell?
Should this work already and if so are there any limitations regarding e.g. X11 and Wayland?

**Comment by stuartmorgan** [comment]
> What has to be done for this to work now that the gtk based shell is implemented as far as I can tell?

The GTK embedding would need to be updated to pass touch events to the engine as `kFlutterPointerDeviceKindTouch` events.

**Comment by RBT22** [comment]
Any updates on this?
Is it difficult to implement it, or is it just not a priority?

**Comment by stuartmorgan** [comment]
Please see https://github.com/flutter/flutter/wiki/Issue-hygiene#when-will-my-bug-be-fixed and https://github.com/flutter/flutter/wiki/Issue-hygiene#do-not-add-me-too-or-same-or-is-there-an-update-comments-to-bugs.

Anyone interested in this functionality is welcome to submit a PR for it. How difficult it is would be something that would be determined while implementing it.

**Comment by thilojaeggi** [comment]
Is there still no update on this? I'm facing the issue when tapping a button the hover state stays until i tap another button cause it seems like it handles it like a mouse.
